webapp/app.py

```diff
@@ -73,11 +73,28 @@ def thank_you():
 
 @open_id.after_login
 def after_login(resp):
-    flask.session["macaroon_discharge"] = resp.extensions["macaroon"].discharge
-    flask.session["openid"] = {
-        "identity_url": resp.identity_url,
-        "email": resp.email,
+    root = flask.session["macaroon_root"]
+    discharge = resp.extensions["macaroon"].discharge
+    authorization_header = get_authorization_header(root, discharge)
+    authorization_code = f"root={root} discharge={discharge}"
+    invitation_code = flask.session["invitation_code"]
+    data = {
+        "provider": "usso",
+        "authorization_code": authorization_code,
+        "invitation_code": invitation_code,
     }
+
+    response = requests.post(
+        url=f"{ANBOXCLOUD_API_BASE}1.0/login",
+        headers=authorization_header,
+        json=data,
+    )
+    body = response.json()
+    if response.status_code == 401:
+        return flask.render_template("401.html", error=body["error"])
+    flask.session.clear()
+    flask.session["authentication_token"] = body["metadata"]["token"]
+
     return flask.redirect(open_id.get_next_url())
```

Comment:
This is not redirecting to `/demo` when hit from `/login`

Reply:
Is that because it doesn't have a `?next_url` parameter or whatever it needs?

Reply:
Just tested this, it doesn't work. We are still unable to redirect user to `/demo` after login